src/test/java/bio/terra/service/iam/sam/SamIamTest.java

```diff
@@ -57,4 +135,274 @@ public void testExtractErrorMessageDeepNested() {
+        final List<IamRole> allPolicies = Arrays.asList(
+            IamRole.ADMIN,
+            IamRole.STEWARD,
+            IamRole.CUSTODIAN,
+            IamRole.SNAPSHOT_CREATOR
+        );
+        final List<IamRole> syncedPolicies = Arrays.asList(
+            IamRole.STEWARD,
+            IamRole.CUSTODIAN,
+            IamRole.SNAPSHOT_CREATOR
+        );
+
+        for (IamRole policy : allPolicies) {
+            when(samGoogleApi.syncPolicy(
+                IamResourceType.DATASET.getSamResourceName(),
+                datasetId.toString(),
+                policy.toString())
+            ).thenReturn(Maps.newHashMap("policygroup-" + policy + "@firecloud.org", Collections.emptyList()));
+        }
+
+        assertThat(samIam.createDatasetResource(userReq, datasetId))
+            .isEqualTo(syncedPolicies.stream().collect(Collectors.toMap(
+                p -> p,
+                p -> "policygroup-" + p + "@firecloud.org"
+            )));
+    }
+
+    @Test
+    public void testCreateSnapshot() throws InterruptedException, ApiException {
+        final UUID snapshotId = UUID.randomUUID();
+        // Note: in our case, policies have a 1:1 relationship with roles
+        final List<IamRole> allPolicies = Arrays.asList(
+            IamRole.ADMIN,
+            IamRole.STEWARD,
+            IamRole.READER,
+            IamRole.DISCOVERER
+        );
+        final List<IamRole> syncedPolicies = Arrays.asList(
+            IamRole.STEWARD,
+            IamRole.READER
+        );
+
+        for (IamRole policy : allPolicies) {
+            when(samGoogleApi.syncPolicy(
+                IamResourceType.DATASNAPSHOT.getSamResourceName(),
+                snapshotId.toString(),
+                policy.toString())
+            ).thenReturn(Maps.newHashMap("policygroup-" + policy + "@firecloud.org", Collections.emptyList()));
+        }
+
+        assertThat(samIam.createSnapshotResource(userReq, snapshotId, Collections.emptyList()))
+            .isEqualTo(syncedPolicies.stream().collect(Collectors.toMap(
+                p -> p,
+                p -> "policygroup-" + p + "@firecloud.org"
+            )));
+    }
+
+    @Test
+    public void testCreateProfile() throws InterruptedException, ApiException {
+        final UUID profileId = UUID.randomUUID();
+        // Note: in our case, policies have a 1:1 relationship with roles
+        final List<IamRole> allPolicies = Arrays.asList(
+            IamRole.ADMIN,
+            IamRole.OWNER,
+            IamRole.USER
+        );
+
+        for (IamRole policy : allPolicies) {
+            when(samGoogleApi.syncPolicy(
+                IamResourceType.DATASNAPSHOT.getSamResourceName(),
+                profileId.toString(),
+                policy.toString())
+            ).thenReturn(Maps.newHashMap("policygroup-" + policy + "@firecloud.org", Collections.emptyList()));
+        }
+
+        samIam.createProfileResource(userReq, profileId.toString());
```

Comment:
Do you need to verify this result somehow?

Reply:
Ah good eye.  Yeah, I'll add a verify

Reply:
Oh wait, now I remember....this method returns a void, which is why I didn't.  I just checked to make sure that it successfully files

Reply:
Gotcha!